src/main/java/com/ldtteam/structurize/client/BlueprintHandler.java

```diff
@@ -139,22 +119,17 @@ public void drawAtListOfPositions(final BlueprintPreviewData previewData,
 
         Minecraft.getInstance().getProfiler().push("struct_render_multi");
 
-        final int blueprintHash = previewData.getBlueprint().hashCode();
-        final BlueprintRenderer rendererRef = rendererCache.get(blueprintHash);
-        final BlueprintRenderer renderer = rendererRef == null ? BlueprintRenderer.buildRendererForBlueprint(previewData.getBlueprint()) : rendererRef;
-
-        if (rendererRef == null)
-        {
-            rendererCache.put(blueprintHash, renderer);
-        }
+        final BlueprintRenderer renderer = rendererCache.getUnchecked(previewData.getRenderKey());
 
         for (final BlockPos coord : points)
         {
             renderer.draw(previewData, coord, ctx);
         }
 
-        evictTimeCache.put(blueprintHash, System.currentTimeMillis());
-
         Minecraft.getInstance().getProfiler().pop();
     }
+
+    public record RenderingCacheKey(RotationMirror rotationMirror, Blueprint blueprint)
```

Comment:
could use some doc as well and {} same line maybe?

Reply:
we do {} new line, so i keep it that way